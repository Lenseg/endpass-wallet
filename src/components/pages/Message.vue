<template>
  <div class="app-page">
    <div class="section">
      <div class="container is-narrow">
        <div class="card app-card main-app-card">
          <div class="card-header">
            <h1 class="card-header-title">Message</h1>
          </div>
          <div class="card-content">
            <div class="tabs">
              <ul>
                <li :class="[ isSignTabActive ? 'is-active' : '']">
                  <a @click="setSignTabActive">Sign</a>
                </li>
                <li :class="[ isVerifyTabActive ? 'is-active' : '']">
                  <a @click="setVerifyTabActive">Verify</a>
                </li>
              </ul>
            </div>
            <sign-message v-show="isSignTabActive"/>
            <verify-message v-show="isVerifyTabActive"/>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import SignMessage from '@/components/SignMessage';
import VerifyMessage from '@/components/VerifyMessage';

const ACTION_TYPES = {
  SIGN: 'SIGN',
  VERIFY: 'VERIFY'
};

export default {
  name: 'message-page',
  data: () => ({
    action: ACTION_TYPES.SIGN
  }),
  computed: {
    isSignTabActive() {
      return this.action === ACTION_TYPES.SIGN;
    },
    isVerifyTabActive() {
      return this.action === ACTION_TYPES.VERIFY;
    }
  },
  methods: {
    setSignTabActive() {
      this.action = ACTION_TYPES.SIGN;
    },
    setVerifyTabActive() {
      this.action = ACTION_TYPES.VERIFY;
    }
  },
  components: {
    SignMessage,
    VerifyMessage
  }
};
</script>
