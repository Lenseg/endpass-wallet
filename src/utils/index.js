export const kebabToCamel = input =>
  input.replace(/-([a-z])/g, g => g[1].toUpperCase());
