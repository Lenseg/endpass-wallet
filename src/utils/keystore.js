import keythereum from 'keythereum';
import bs58check from 'bs58check';
import { kdfParams } from '@/config';

// Monkey patch keythereum to skip generating address for private keys
// This allows us to encrypt private keys of arbitrary length, and
// conforms better to the Ethereum keystore V3 spec, which does not include
// the address for privacy reasons.
// See https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#alterations-from-version-1
keythereum.privateKeyToAddress = function(pk) {
  return '';
};

export default {
  // Encrypts a private key Buffer into a V3 keystore object
  // The exported keystore does NOT include an address
  encrypt(password, privateKey) {
    // Generate random salt and iv for each encryption
    let dk = keythereum.create();
    let options = {
      kdf: kdfParams.kdf,
      kdfparams: kdfParams,
    };
    let encrypted = keythereum.dump(
      password,
      privateKey,
      dk.salt,
      dk.iv,
      options,
    );
    delete encrypted.address;
    return encrypted;
  },

  // Decrypts a V3 keystore object into a private key Buffer
  decrypt(password, keystore) {
    return keythereum.recover(password, keystore);
  },

  // Encode to Base58Check
  encodeBase58(key) {
    return bs58check.encode(key);
  },

  // Decode from Base58Check
  decodeBase58(key) {
    return bs58check.decode(key);
  },
};
