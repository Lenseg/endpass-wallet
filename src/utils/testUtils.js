const toKebab = string =>
  string.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();

export function generateStubs(Component) {
  return Object.values(Component.components).reduce((stubs, stubComponent) => {
    let dashName;

    if (stubComponent.name) {
      dashName = toKebab(stubComponent.name);
    }

    stubs[dashName] = {
      render(createElement) {
        return createElement(dashName, this.$slots.default);
      },
    };

    return stubs;
  }, {});
}
