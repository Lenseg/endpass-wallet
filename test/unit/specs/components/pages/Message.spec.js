import { shallow } from '@vue/test-utils';
import Message from '@/components/pages/Message';
import { generateStubs } from '@/utils/testUtils';

describe('Message page', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(Message, {
      stubs: generateStubs(Message)
    });
  });

  describe('render', () => {
    it('should render the initial state of the page', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
