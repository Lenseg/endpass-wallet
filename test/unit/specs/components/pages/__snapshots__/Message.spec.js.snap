// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message page render should render the initial state of the page 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          >
            Message
          </h1>
        </div>
         
        <div
          class="card-content"
        >
          <div
            class="tabs"
          >
            <ul>
              <li
                class="is-active"
              >
                <a>
                  Sign
                </a>
              </li>
               
              <li
                class=""
              >
                <a>
                  Verify
                </a>
              </li>
            </ul>
          </div>
           
          <sign-message />
           
          <verify-message
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message page render should set Sign tab 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          >
            Message
          </h1>
        </div>
         
        <div
          class="card-content"
        >
          <div
            class="tabs"
          >
            <ul>
              <li
                class="is-active"
              >
                <a>
                  Sign
                </a>
              </li>
               
              <li
                class=""
              >
                <a>
                  Verify
                </a>
              </li>
            </ul>
          </div>
           
          <sign-message />
           
          <verify-message
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message page render should set Verify tab 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          >
            Message
          </h1>
        </div>
         
        <div
          class="card-content"
        >
          <div
            class="tabs"
          >
            <ul>
              <li
                class=""
              >
                <a>
                  Sign
                </a>
              </li>
               
              <li
                class="is-active"
              >
                <a>
                  Verify
                </a>
              </li>
            </ul>
          </div>
           
          <sign-message
            style="display: none;"
          />
           
          <verify-message
            style=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
